app/src/main/java/com/sdp/movemeet/ProfileActivity.java

```diff
@@ -176,64 +176,76 @@ public void goToEditProfileActivity(View view) {
 
 
     public void deleteUserAccount(View view) {
+        // 1) Delete the profile picture of the user from Firebase Storage (in case it exists)
+        profileRef = storageReference.child(userImagePath);
+        profileRef.getDownloadUrl().addOnSuccessListener(new OnSuccessListener<Uri>() {
```

Comment:
Having those subsequent callbacks looks a bit confusing, maybe you can refactor this ?
For example, you could do:
profileRef.getDownloadUrl().addOnSuccessListener(new OnSuccessListener<Uri>() {  
            @Override
            public void onSuccess(Uri uri) { 
              deleteProfilePicture(profileRef);
           }
}

and in deleteProfilePicture(), you would have the callback to delete the user data. It would look cleaner imo

Reply:
That's a good idea! Thank you for this recommendation Victor, it would in fact look less confusing!